docs/manual/file_mode.md

```diff
@@ -0,0 +1,95 @@
+---
+layout: default
+title: rpm.org - File Mode
+---
+# File Mode
+
+RPM supports a variety of compression methods through external libraries for
+its internal I/O operations.  For ease of use, the [RPMIO
+API](https://ftp.osuosl.org/pub/rpm/api/4.17.0/group__rpmio.html) offers
+special `fopen(3)/fdopen(3)`-like functions that accept an extended *mode
+string* which allows for specifying a compression method alongside the usual
+`r` or `w` letters.  RPM will then perform (de)compression transparently for
+the caller.
+
+While mostly used internally, a mode string can also be exposed through
+configuration options or macros where appropriate.  Currently, only payload
+write mode can be configured this way, see the [Payload](format.html#payload)
+section for details.
+
+## Format
+
+The format of an RPMIO mode string is as follows:
+
+```
+<mode>[flags].<type>
+```
+
+The `mode` parameter is mandatory and contains a sequence of letters as
+described in `fopen(3)`.  For payload configuration, only `w` is allowed.
+
+The `type` parameter is also mandatory and specifies the compression type
+(*compressor*) to be used.  Currently, the following compressors are supported:
+
+| Type		| Description					    |
+|---------------|---------------------------------------------------|
+|`gzdio`	| gzip ([flags](#gzip-flags))			    |
+|`bzdio`	| bzip2 ([flags](#bzip2-flags))			    |
+|`xzdio`	| lzma ([flags](#xz-lzma-flags))		    |
+|`lzdio`	| lzma alone (legacy) ([flags](#lzma-alone-flags))  |
+|`zstdio`	| zstd ([flags](#zstd-flags))			    |
+|`ufdio`	| uncompressed					    |
+
+The `flags` parameter is optional and contains compressor-specific options,
+most commonly a compression *level* or the *number of threads*.  For the actual
+supported values, consult the given compressor's section below.  For some
+real-world examples, see [Example Modes](#example-modes).
+
+#### GZIP Flags
+
+These can include a compression level (from 0 to 9) and a strategy (e.g. `h`
+for Huffman-only compression).  See the description of `gzopen()` in
+`/usr/include/zlib.h` for more details.
+
+#### BZIP2 Flags
+
+These can include a compression level (from 1 to 9) and `s` (enables "small
+mode").
+
+#### XZ (LZMA) Flags
+
+These can include a compression level (from 0 to 9) and `T` followed by the
+number of threads to be used during compression, where `T0` (or just `T`) means
+that the optimal number of threads should automatically be chosen based on the
+`%{getncpus}` macro.
+
+#### LZMA (Alone) Flags
+
+These can include a compression level (from 0 to 9).
+
+#### ZSTD Flags
+
+These can include a compression level (see `-#` in `zstd(1)` for the supported
+values), `T` followed by the number of threads to be used during compression,
+where `T0` means that the optimal number of threads should automatically be
+chosen based on the `%{getncpus}` macro, and `L` followed by a window size that
```

Comment:
Ditto here, just say number of threads should be automatically detected, but don't mention how exactly.